packages/vue-apollo/src/mixin.js

```diff
@@ -14,15 +14,6 @@ function initProvider () {
       : optionValue
   } else if (options.parent && options.parent.$apolloProvider) {
     this.$apolloProvider = options.parent.$apolloProvider
-  } else if (options.provide) {
-    // TODO remove
-    // Temporary retro-compatibility
-    const provided = typeof options.provide === 'function'
-      ? options.provide.call(this)
-      : options.provide
-    if (provided && provided.$apolloProvider) {
-      this.$apolloProvider = provided.$apolloProvider
-    }
```

Comment:
**Heads up!** I haven't tested this out thoroughly and there might be some unintended consequences with removing this.

But removing this bit, fixed our immediate problem and it looks like it was slated for deletion anyways 😅